spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java

```diff
@@ -245,6 +246,15 @@ private void configureProperties(FluentConfiguration configuration, FlywayProper
 			// No method reference for compatibility with Flyway 6.x
 			map.from(properties.getSkipExecutingMigrations()).whenNonNull()
 					.to((skipExecutingMigrations) -> configuration.skipExecutingMigrations(skipExecutingMigrations));
+			// Teams secrets management properties (all non-method reference for
+			// compatibility with Flyway 6.x)
+			map.from(properties.getConjurUrl()).whenNonNull().to((conjurUrl) -> configuration.conjurUrl(conjurUrl));
```

Comment:
```suggestion
			map.from(properties.getConjurUrl()).whenNonNull().to(configuration::conjurUrl);
```

Reply:
The method reference does not work when executing against the 5.x/6.x older `FluentConfiguration` which does not have the `conjur**` or `vault**` mutators. Regardless of the value being null, at runtime the method reference still attempts to get resolved at which time a NoSuchMethod gets thrown. This is why the file is littered w/ comments like this ```No method reference for compatibility with Flyway 6.x```. 

To see this in action run `org.springframework.boot.autoconfigure.flyway.Flyway5xAutoConfigurationTests` or `org.springframework.boot.autoconfigure.flyway.Flyway6xAutoConfigurationTests`.